### PR TITLE
Wy/blur pool

### PR DESCRIPTION
**Pull request by yikwill**
# Modulus Pull Request
Adds BlurPool for dealiased downsampling as in [this paper](https://proceedings.mlr.press/v97/zhang19a/zhang19a.pdf).

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

### CODE REVIEW

File: physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py [hunk 1]
```diff
@@ -1010,6 +1013,91 @@ def forward(self, x):
         """
         return self.avgpool(x)
 
+class BlurPool(th.nn.Module):
```
Comment (akshaysubr) [inline physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py]: Would prefer referring to this as `DealiasedDownsample` instead even though `BlurPool` is the specific way of doing this here.

File: physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py [hunk 2]
```diff
@@ -1010,6 +1013,91 @@ def forward(self, x):
...
+            groups=in_channels,
+            bias=False,
+            dilation=1,
+            enable_nhwc=enable_nhwc,
+            enable_healpixpad=enable_healpixpad,
+        )
+        w = BlurPool._normalized_depthwise_blur_weights(filt, in_channels)
+        if enable_nhwc:
+            w = w.to(memory_format=th.channels_last)
+        conv = self.pool.layers[-1]
```
Comment (akshaysubr) [inline physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py]: This is a bit brittle isn't it and susceptible to internal changes within `geometry_layer`?

File: physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py [hunk 3]
```diff
@@ -1010,6 +1013,91 @@ def forward(self, x):
...
+            raise ValueError("resample_filter must not sum to zero")
+
+        # Use torch.nn.Conv2d as the layer class so HEALPixLayer recognizes
+        # it as a convolution layer and adds HEALPixPadding
+        self.pool = geometry_layer(
+            layer=th.nn.Conv2d,
+            in_channels=in_channels,
+            out_channels=in_channels,
+            kernel_size=m,
+            stride=stride,
```
Comment (akshaysubr) [inline physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py]: Is this always correct? With the default resample filter, this works with stride=2. But if you want stride=4, don't you need to use resample_filter convolved with itself? May be good to require stride be a power of 2 and have log(stride) pool layers?

File: physicsnemo/models/dlwp_healpix_layers/healpix_layers.py [hunk 4]
```diff
@@ -582,16 +582,16 @@ def __init__(self, layer, **kwargs):
             del kwargs["enable_healpixpad"]
         else:
             enable_healpixpad = False
+        
+        kernel_size = 3 if "kernel_size" not in kwargs else kwargs["kernel_size"]
```
Comment (akshaysubr) [inline physicsnemo/models/dlwp_healpix_layers/healpix_layers.py:586]: What happens here if we use the binomial resample filter which has a kernel size of 5?

**Review by akshaysubr — COMMENTED** [review]
The new changes look good to me. I had one comment that might be critical to resolve though.

File: physicsnemo/models/dlwp_healpix_layers/healpix_layers.py (hunk 4 above)
Comment (yikwill) [reply physicsnemo/models/dlwp_healpix_layers/healpix_layers.py:586]: We pass `kernel_size=m` where `m` is the length of the filter given to `DealiasedDownsample`, so this will be treated as a size 5 kernel and the appropriate padding of 2 will be added by `HEALPixLayer`. I believe this behavior is desired since we'd like the hpx pad the input before applying our dealiasing filter.

**Review by akshaysubr — APPROVED** [review]
